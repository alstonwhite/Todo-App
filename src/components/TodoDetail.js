import React from "react";

const TodoDetail = ({ active, onUpdateTask, onComplete, onDelete }) => {
  return (
    <div className="app-column">
      <div className="header task-header">
        <input
          className="title-input"
          type="text"
          value={active ? active.title : ""}
          onChange={event => onUpdateTask(active, event.target.value)}
        />
        <button
          className="button button-large button-complete"
          onClick={() => onComplete(active)}
        >
          Complete
        </button>{" "}
        <button
          className="button button-large button-delete"
          onClick={() => onDelete(active)}
        >
          Remove
        </button>{" "}
      </div>
      <div className="task-body">
        <textarea
          className="body-text"
          type="text"
          value={active ? active.body : ""}
          onChange={event =>
            onUpdateTask(active, undefined, event.target.value)
          }
        />
      </div>
    </div>
  );
};

export default TodoDetail;
