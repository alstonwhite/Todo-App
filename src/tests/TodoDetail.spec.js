import React from "react";
import renderer from "react-test-renderer";
import ReactTestUtils from 'react-dom/test-utils';

import TodoDetail from "../components/TodoDetail";

it("Renders a task correctly", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "new",
        body: "Add notes..."
      }
    ];
    const updateTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoDetail
          active={item}
          onUpdateTask={updateTask}
          onComplete={completeTask}
          onDelete={deleteTask}
          />
      );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("Complete task button triggers function on button click", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "completed",
        body: "Add notes..."
      }
    ];
    const updateTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoDetail
          active={item}
          onUpdateTask={updateTask}
          onComplete={completeTask}
          onDelete={deleteTask}
          />
      );
    let button = tree.root.findByProps({className:'button button-large button-complete'});
    ReactTestUtils.Simulate.click(button);
    expect(completeTask).toHaveBeenCalled();
  });

  it("Delete task button triggers function on button click", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "completed",
        body: "Add notes..."
      }
    ];
    const updateTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoDetail
          active={item}
          onUpdateTask={updateTask}
          onComplete={completeTask}
          onDelete={deleteTask}
          />
      );
    let button = tree.root.findByProps({className:'button button-large button-delete'});
    ReactTestUtils.Simulate.click(button);
    expect(deleteTask).toHaveBeenCalled();
  });

  it("Title input triggers function on change", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "completed",
        body: "Add notes..."
      }
    ];
    const updateTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoDetail
          active={item}
          onUpdateTask={updateTask}
          onComplete={completeTask}
          onDelete={deleteTask}
          />
      );
    let button = tree.root.findByProps({className:'title-input'});
    ReactTestUtils.Simulate.change(button);
    expect(updateTask).toHaveBeenCalled();
  });

  it("Body textarea triggers function on change", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "completed",
        body: "Add notes..."
      }
    ];
    const updateTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoDetail
          active={item}
          onUpdateTask={updateTask}
          onComplete={completeTask}
          onDelete={deleteTask}
          />
      );
    let button = tree.root.findByProps({className:'body-text'});
    ReactTestUtils.Simulate.change(button);
    expect(updateTask).toHaveBeenCalled();
  });



// Renders: Item (with new/completed status?)
// Complete button: Calls proper fn
// Delete button: Calls proper fn
// Title input: Calls proper fn
// Body input: Calls proper fn