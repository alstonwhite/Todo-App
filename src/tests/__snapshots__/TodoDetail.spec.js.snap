// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a task correctly 1`] = `
<div
  className="app-column"
>
  <div
    className="header task-header"
  >
    <input
      className="title-input"
      onChange={[Function]}
      type="text"
    />
    <button
      className="button button-large button-complete"
      onClick={[Function]}
    >
      Complete
    </button>
     
    <button
      className="button button-large button-delete"
      onClick={[Function]}
    >
      Remove
    </button>
     
  </div>
  <div
    className="task-body"
  >
    <textarea
      className="body-text"
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;
